backend/functions/lib/adapters/gateways/user-session-cache.cosmosdb.repository.ts

```diff
@@ -0,0 +1,71 @@
+import { ApplicationContext } from '../types/basic';
+import { CosmosDbRepository } from './cosmos/cosmos.repository';
+import { CamsSession } from '../../../../../common/src/cams/session';
+import { UserSessionCacheRepository } from './user-session-cache.repository';
+import { CamsJwtClaims } from '../types/authorization';
+import { UnauthorizedError } from '../../common-errors/unauthorized-error';
+
+const MODULE_NAME: string = 'COSMOS_DB_REPOSITORY_USER_SESSION_CACHE';
+const CONTAINER_NAME: string = 'user-session-cache';
+
+export type CachedCamsSession = CamsSession & {
+  id?: string;
+  signature: string;
+  ttl: number;
+};
+
+export class UserSessionCacheCosmosDbRepository implements UserSessionCacheRepository {
+  private repo: CosmosDbRepository<CachedCamsSession>;
+  constructor(context: ApplicationContext) {
+    this.repo = new CosmosDbRepository<CachedCamsSession>(context, CONTAINER_NAME, MODULE_NAME);
+  }
+
+  public async get(context: ApplicationContext, token: string): Promise<CamsSession | null> {
+    const tokenParts = token.split('.');
+    if (tokenParts.length !== 3) {
+      throw new UnauthorizedError(MODULE_NAME, { message: 'Invalid token received.' });
+    }
+    const signature = tokenParts[2];
+    const query = 'SELECT * FROM c WHERE c.signature = @signature';
+    const querySpec = {
+      query,
+      parameters: [
+        {
+          name: '@signature',
+          value: signature,
+        },
+      ],
+    };
+
+    const cached = await this.repo.query(context, querySpec);
+    if (cached && cached.length === 1) {
+      return toCamsSession(cached[0]);
+    } else {
+      return null;
+    }
+  }
+
+  public async put(context: ApplicationContext, session: CamsSession): Promise<CamsSession> {
+    const tokenParts = session.apiToken.split('.');
+    if (tokenParts.length !== 3) {
+      throw new UnauthorizedError(MODULE_NAME, { message: 'Invalid token received.' });
+    }
+    const signature = tokenParts[2];
+    const tokenBody = tokenParts[1];
+    const claims = JSON.parse(
+      Buffer.from(tokenBody, 'base64').toString(),
+    ) as unknown as CamsJwtClaims;
```

Comment:
Consider using a humble object implementation of the new jwt package we brought in.

Reply:
The jwt library cannot be used in FE code, so it cannot be used in common.